### ISSUE TITLE
Dash playback isnt working in html5

### ISSUE THREAD

**Issue by XeiXion** [YePpHa/YouTubeCenter#1552]
i checked the stats for nerd and it shows up as DASH: yes (243/140) but it still loads the entire video.

i want dash because i usually switch to another video if i don't like it and i have download limit so a lot of data would be wasted if i loaded the entire video and just watched like a minute or two. it works on the flash player but my flash is having some problems with youtube idk why

**Comment by Zarggg** [comment]
What do you mean by "isn't working"? For example, does the video just stop playing after a certain point?

What browser are you using and what version of YTC do you have installed?

**Comment by Zarggg** [comment]
Enabling or disabling DASH doesn't directly control whether a video buffers or not. You want to enable "Prevent auto-buffering" if you want to stop your browser from loading a video while it is not being played.

As far as I know, there is no function in YTC (or YouTube) to do what you want.

**Comment by XeiXion** [comment]
well in the flash player with dash was on it was like while paused it will buffer to a certain point and only buffer further while the video is playing 

Example: playing video > keeps buffering to a limit and then to the next >  paused > stops buffering when it reaches that limit

so while playing it wont load the entire video only like it will keep a 1 minute or something margin of buffer always 

idk if i explained it well :/

**Comment by ErdeFB** [comment]
It always has buffered at least somewhat even if DASH has been enabled as far as I recall.
You can try setting custom buffer settings from Player -> General and see if that works for you.

**Comment by XeiXion** [comment]
custom buffer is doing nothing for me 
if i dint explain it well i made a 20 sec video showing how it was
https://www.youtube.com/watch?v=aQhx6h5QP64&feature=youtu.be

**Comment by allbetsareoff29** [comment]
I have the opposite problem. Newest Dev build. Dash=yes. HTML5. custom buffer=yes. slider 90% to the right.  

video only buffers 20 seconds (as per dash) but then stops and does not buffer to the specified size by the buffer slider

**Comment by Leonick** [comment]
Same as OP. Dash playback doesn't work with HTML5 player. Only gets worse as prevent auto-buffer and custom buffer size doesn't work either...

**Comment by ErdeFB** [comment]
Like I said, even when you have DASH enabled, it still buffers some of the video instead of all of it like was the case in non-DASH videos.

Anyway, if you indeed have set DASH enabled and still your whole video gets buffered, I don't really know what would cause it. Just out of curiosity, what do you see when you go to youtube.com/html5 ?

**Comment by Leonick** [comment]
OK, loaded a video and just let it sit there to see how much it loaded. It didn't actually load the whole video, but it loaded way more than DASH usually does. Out of a 16 minute video it loaded about 7 minutes and 30 seconds and I don't have custom buffer on or anything.

"Stats for nerds" for the video says DASH is on, "yes (135/140)" and it's enabled in youtube center as well.

As for youtube.com/html5
![capture](https://cloud.githubusercontent.com/assets/2903631/7206718/6db20760-e534-11e4-92ba-b40c8565850a.JPG)

**Comment by allbetsareoff29** [comment]
‎When I was messing around, I noticed that the amount it buffers is directly proportional to the video quality/resolution. When selected html5 at 4k results, it buffered 20 seconds and then stopped buffering. At 1440p, the buffer was exactly double - 40 seconds. At 1080p, it buffered 80 seconds...and so on. This makes me believe that the size of the buffer is NOT time based but rather size based. The smaller the resolution, the more time it buffers because at lower resolution, it can buffer "more seconds" to squeeze in the same allotted megabyte limit.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          From: Hampus JensenSent: Friday, April 17, 2015 1:05 PMTo: YePpHa/YouTubeCenterReply To: YePpHa/YouTubeCenterCc: allbetsareoff29Subject: Re: [YouTubeCenter] Dash playback isnt working in html5 (#1552)OK, loaded a video and just let it sit there to see how much it loaded. It didn't actually load the whole video, but it loaded way more than DASH usually does. Out of a 16 minute video it loaded about 7 minutes and 30 seconds and I don't have custom buffer on or anything.

"Stats for nerds" for the video says DASH is on, "yes (135/140)" and it's enabled in youtube center as well.

As for youtube.com/html5


—Reply to this email directly or view it on GitHub.

**Comment by Leonick** [comment]
Makes sense I guess... Wonder why there's such a big difference between HTML5 and Flash though, because with flash it generally doesn't even buffer a minute ahead on 480p, compare that to over 7 minutes. Suppose that's a youtube issue but then again, youtube center is all about sorting out their issues.

**Comment by ErdeFB** [comment]
Just a hunch, but it could be actually due to limitations in the HTML5 players why the system differs. Don't quote me on that though.

**Comment by allbetsareoff29** [comment]
Html5 videos get handled by browser. Flash videos don't. For all intensive purposes, with flash, browser just creates an empty box and handles off the responsibility to the flash plugin. ‎Flash plugin works on its own to "fill" the box created by browser with images/video                                                                                                                                                                                                                                                                                                                                        Sent from my smartphone. Please excuse any typos                                                                                                                                                                                                                From: Hampus JensenSent: Friday, April 17, 2015 1:27 PMTo: YePpHa/YouTubeCenterReply To: YePpHa/YouTubeCenterCc: allbetsareoff29Subject: Re: [YouTubeCenter] Dash playback isnt working in html5 (#1552)Makes sense I guess... Wonder why there's such a big difference between HTML5 and Flash though, because with flash it generally doesn't even buffer a minute ahead on 480p, compare that to over 7 minutes. Suppose that's a youtube issue but then again, youtube center is all about sorting out their issues.

—Reply to this email directly or view it on GitHub.

**Comment by allbetsareoff29** [comment]
Interesting thought. I'd be interested to see how different browsers handle the differences                                                                                                                                                                                                                From: ErdeFBSent: Friday, April 17, 2015 1:33 PMTo: YePpHa/YouTubeCenterReply To: YePpHa/YouTubeCenterCc: allbetsareoff29Subject: Re: [YouTubeCenter] Dash playback isnt working in html5 (#1552)Just a hunch, but it could be actually due to limitations in the HTML5 players why the system differs. Don't quote me on that though.

—Reply to this email directly or view it on GitHub.

**Comment by Leonick** [comment]
Just tried in Chrome, it stopped buffering after 2 and a half minute. 
Firefox only recently had the media source extensions implemented 
(required fof DASH with HTML5 videos) so maybe it's still missing 
something. Though you'd figured they'd implement it properly when they did.

**Comment by allbetsareoff29** [comment]
Try same browser and different video resolutions                                                                                                                                                                                                                From: Hampus JensenSent: Friday, April 17, 2015 1:49 PMTo: YePpHa/YouTubeCenterReply To: YePpHa/YouTubeCenterCc: allbetsareoff29Subject: Re: [YouTubeCenter] Dash playback isnt working in html5 (#1552)Just tried in Chrome, it stopped buffering after 2 and a half minute. 
Firefox only recently had the media source extensions implemented 
(required fof DASH with HTML5 videos) so maybe it's still missing 
something. Though you'd figured they'd implement it properly when they did.


—Reply to this email directly or view it on GitHub.

**Comment by Leonick** [comment]
At 360p it firefox loaded 10 minutes 44 seconds of the same 16 minute video.

**Comment by RoboGigas** [comment]
I've experienced an issue with Dash playback that mainly seems to affect music videos (no real video, just audio) where the playback bar appears to buffer up to a minute but the video is unable to start regardless of what you try, seemingly buffering forever. Turning off Dash playback enables the videos to render, but I can't keep changing this option to go between having 60fps or the videos buffering properly.
